File: jewelry-store/src/app/services/comment.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Comment } from '../models/comment'; // ✅ pravi model

@Injectable({
  providedIn: 'root'
})
export class CommentService {
  private apiUrl = 'https://localhost:7069/api/comments';

  constructor(private http: HttpClient) {}

  getComments(productId: number): Observable<Comment[]> {
    return this.http.get<Comment[]>(`${this.apiUrl}/product/${productId}`);
  }

  addComment(comment: Partial<Comment>): Observable<Comment> {
    const token = localStorage.getItem('token');
    const headers = new HttpHeaders({ Authorization: `Bearer ${token}` });

    return this.http.post<Comment>(this.apiUrl, comment, { headers });
  }
}

File: jewelry-store-api/JewelryStoreApi/Models/Comment.cs
using System;

namespace JewelryStoreApi.Models
{
    public class Comment
    {
        public int? Id { get; set; }
        public int? ProductId { get; set; }
        public Product? Product { get; set; }
        public string? UserName { get; set; }
        public string? Text { get; set; }
        public DateTime? CreatedAt { get; set; } = DateTime.UtcNow;
    }
}
